xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -125,11 +138,18 @@ func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (Clu
 		}
 	}
 
+	var od *OutlierDetection
```

Comment:
Any reason for not protecting this with the env var `GRPC_EXPERIMENTAL_ENABLE_OUTLIER_DETECTION` as defined in the gRFC?

Reply:
I was going to do it in a separate PR, but yeah it makes sense to go ahead and do it here. Added.